### PR TITLE
[WIP] Require fully authenticated to change passwords

### PR DESCRIPTION
**Pull request by leofeyer**
This PR implements #1273.

- [ ] Do we need the check in the "close account" module?
- [ ] Do we really want to show the login module where the member could log in as a different user?

### CODE REVIEW

File: src/Resources/contao/modules/ModuleChangePassword.php [hunk 1]
```diff
@@ -53,6 +54,12 @@ public function generate()
 			return '';
 		}
 
+		// Require the member to be fully authenticated (see #1273)
+		if (!\System::getContainer()->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY'))
+		{
+			throw new AccessDeniedException('Please authenticate');
```
Comment (aschempp) [inline src/Resources/contao/modules/ModuleChangePassword.php]: As this message is not shown to the user, I think it should rather be something like `Full authentication is required to change the password`.

File: src/Resources/contao/modules/ModuleCloseAccount.php [hunk 2]
```diff
@@ -53,6 +54,12 @@ public function generate()
 			return '';
 		}
 
+		// Require the member to be fully authenticated (see #1273)
+		if (!\System::getContainer()->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY'))
+		{
+			throw new AccessDeniedException('Please authenticate');
```
Comment (aschempp) [inline src/Resources/contao/modules/ModuleCloseAccount.php]: Same here

**Comment by leofeyer** [comment]
> Do we need to be fully authenticated for the password change module? It requires to enter the current password anyway.

That's a very good point. And since it also requires to enter the password to close an account, we might not need this check at all.

**Comment by ausi** [comment]
How does this work with two factor authentication?

**Comment by aschempp** [comment]
> How does this work with two factor authentication?

We don't have two-factor authentication in the front end (yet)

**Pull request closed without merging**